modules/core/04-channel/v2/types/merkle.go

```diff
@@ -1,31 +1,20 @@
 package types
 
 import (
-	commitmenttypes "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types"
 	commitmenttypesv2 "github.com/cosmos/ibc-go/v9/modules/core/23-commitment/types/v2"
 )
 
 // BuildMerklePath takes the merkle path prefix and an ICS24 path
 // and builds a new path by appending the ICS24 path to the last element of the merkle path prefix.
-func BuildMerklePath(prefix commitmenttypesv2.MerklePath, path []byte) commitmenttypesv2.MerklePath {
-	if prefix.Empty() {
-		return commitmenttypes.NewMerklePath(path)
+func BuildMerklePath(prefix [][]byte, path []byte) commitmenttypesv2.MerklePath {
+	prefixLength := len(prefix)
+	if prefixLength == 0 {
+		panic("cannot build merkle path with empty prefix")
 	}
 
-	// avoid mutating the provided prefix
-	prefixKeys := make([][]byte, len(prefix.KeyPath))
-	copy(prefixKeys, prefix.KeyPath)
-
-	lastElement := prefixKeys[len(prefixKeys)-1]
+	// copy prefix to avoid modifying the original slice
+	prefix = append([][]byte(nil), prefix...)
```

Comment:
Should we name this var to something else since it eventually becomes the full path?

Reply:
Agreed.
Also good to not reassign or change function arguments, so would prefer a new name in any case.

Reply:
done